neuralop/layers/integral_transform.py

```diff
@@ -90,46 +89,49 @@ def __init__(
     the same as the channels of f
     """
 
-    def forward(
-        self,
-        y,
-        neighbors,
-        x=None,
-        f_y=None,
-        weights=None,
-    ):
+    def forward(self, y, neighbors, x=None, f_y=None, weights=None, batched=False):
         """Compute a kernel integral transform
 
         Parameters
         ----------
         y : torch.Tensor of shape [n, d1]
             n points of dimension d1 specifying
             the space to integrate over.
+            If batched, these must remain constant
+            over the whole batch so no batch dim is needed.
         neighbors : dict
             The sets A(x) given in CRS format. The
             dict must contain the keys "neighbors_index"
             and "neighbors_row_splits." For descriptions
             of the two, see NeighborSearch.
+            If batch > 1, the neighbors must be constant
+            across the entire batch.
         x : torch.Tensor of shape [m, d2], default None
             m points of dimension d2 over which the
             output function is defined. If None,
             x = y.
-        f_y : torch.Tensor of shape [n, d3], default None
+        f_y : torch.Tensor of shape [batch, n, d3] or [n, d3], default None
             Function to integrate the kernel against defined
             on the points y. The kernel is assumed diagonal
             hence its output shape must be d3 for the transforms
             (b) or (d). If None, (a) is computed.
-        weights : torch.Tensor of shape [n,], default None
+        weights : torch.Tensor of shape [batch, n] or [n,], default None
             Weights for each point y proprtional to the
             volume around f(y) being integrated. For example,
             suppose d1=1 and let y_1 < y_2 < ... < y_{n+1}
             be some points. Then, for a Riemann sum,
             the weights are y_{j+1} - y_j. If None,
             1/|A(x)| is used.
+        batched: bool, default True
```

Comment:
Why don't we infer this option from the input?

Reply:
That sounds good. Updated